### DISCUSSION TITLE
offline wallet/signing library support

### DISCUSSION THREAD

**Discussion by longfeiWan9** [filecoin-project/community#240] [Developers]
There are community members in the slack channels asking questions about offline wallet creation,  signing and broadcasting messages to the Filecoin network. It seems that we either do not have tools to provide those features, or the instructions/examples are not clear. So I think maybe we should start a discussion here to figure out:

1. Should we provide/improve tools or libraries to offer offline wallet creation (including multisign wallet), message signing and broadcasting, etc? 
2. If any community already has those tools, please let us know here so that we can educate builders or improve our tools.
3. Do we have a clean guide, examples, and demos to teach Filecoin builders? What should we improve?

-----
FYI, questions being asked:

- Does Filecoin support creating multisig wallets offline?
- How can we create a Filecoin wallet in node js without running local node?
- Is there any example or demo to create a raw message and sign it using golang?
- Is there any tutorial or demos on how to create raw transactions, offline signing?
- I cannot find any doc for offline signing transactions and then broadcasting the rawT to the nodes. How to do it in Filecoin?

-----
**Keywords**:
- offline multisig wallet creation
- sign and send transactions/messages offline
- demo/examples
- Docs is outdated
- multi-language supports: golang, js

**Comment by eshon** [comment]
https://www.npmjs.com/package/iso-filecoin is currently the recommended Filecoin signing/wallet tools library, it's actively maintained
